# 1 - Define the problems / Questions

2 -  Data Collections -

3 -  Data Cleaning -
 - Handle missing values
 - Identify and remove duplicates
 - Handle outliers
 - Transform the data into suitable format


4 -  EDA - Exploratory Data Analysis.


5 - Data Visualizations-


6 - Analyze the insights & Report


In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
titanic = sns.load_dataset("titanic")

In [ ]:
# initial data inspection

In [ ]:
titanic.head()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True

In [ ]:
titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   survived     891 non-null    int64   
 1   pclass       891 non-null    int64   
 2   sex          891 non-null    object  
 3   age          714 non-null    float64 
 4   sibsp        891 non-null    int64   
 5   parch        891 non-null    int64   
 6   fare         891 non-null    float64 
 7   embarked     889 non-null    object  
 8   class        891 non-null    category
 9   who          891 non-null    object  
 10  adult_male   891 non-null    bool    
 11  deck         203 non-null    category
 12  embark_town  889 non-null    object  
 13  alive        891 non-null    object  
 14  alone        891 non-null    bool    
dtypes: bool(2), category(2), float64(2), int64(4), object(5)
memory usage: 80.7+ KB


In [ ]:
# As we seen, the data contain some missing values.
# So lets check the exact number

In [ ]:
titanic.isnull().sum()

survived         0
pclass           0
sex              0
age            177
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64

# Insights

 - age -  177 missing values - should be imputed based on pclass or sex - replace by mean or median
 - embarked - 2 missing values -  can be filled with mode
 - deck - massive values -  better to drop this column


# .fillna()

.dropna.dop()

In [ ]:
# fill with median age based on pclass & sex

In [ ]:
titanic['age'] = titanic.groupby(['pclass','sex'])['age'].transform(lambda x : x.fillna(x.median()))

In [ ]:
titanic['age'].isnull().sum()

0

In [ ]:
# replace embark with mode

In [ ]:
titanic['embarked'].mode()[0]

'S'

In [ ]:
titanic['embarked'].fillna(titanic['embarked'].mode()[0],inplace = True)
# inplace = True is for permenent changes

In [ ]:
titanic['embarked'].isnull().sum()

0

In [ ]:
titanic.drop(columns=['deck'],inplace= True)

In [ ]:
titanic.isnull().sum()

survived       0
pclass         0
sex            0
age            0
sibsp          0
parch          0
fare           0
embarked       0
class          0
who            0
adult_male     0
embark_town    2
alive          0
alone          0
dtype: int64

In [ ]:
titanic.drop(columns=['who','embark_town','alive', 'adult_male'], inplace= True)

In [ ]:
titanic.isnull().sum()

survived    0
pclass      0
sex         0
age         0
sibsp       0
parch       0
fare        0
embarked    0
class       0
alone       0
dtype: int64

In [ ]:
titanic.isnull().sum()

survived    0
pclass      0
sex         0
age         0
sibsp       0
parch       0
fare        0
embarked    0
class       0
alone       0
dtype: int64

In [ ]:
titanic.shape

(891, 10)

After dropping unwanted columns the titanic dataset now has 891 rows describe the 891 passanger details and 10 columns which shows the passanger features.

In [ ]:
# lets check for statistical summery of the data

In [ ]:
titanic.describe(include='all')

survived      pclass   sex         age       sibsp       parch  \
count   891.000000  891.000000   891  891.000000  891.000000  891.000000   
unique         NaN         NaN     2         NaN         NaN         NaN   
top            NaN         NaN  male         NaN         NaN         NaN   
freq           NaN         NaN   577         NaN         NaN         NaN   
mean      0.383838    2.308642   NaN   29.112424    0.523008    0.381594   
std       0.486592    0.836071   NaN   13.304424    1.102743    0.806057   
min       0.000000    1.000000   NaN    0.420000    0.000000    0.000000   
25%       0.000000    2.000000   NaN   21.500000    0.000000    0.000000   
50%       0.000000    3.000000   NaN   26.000000    0.000000    0.000000   
75%       1.000000    3.000000   NaN   36.000000    1.000000    0.000000   
max       1.000000    3.000000   NaN   80.000000    8.000000    6.000000   

              fare embarked  class alone  
count   891.000000      891    891   891  
unique         NaN        3      3     2  
top            NaN        S  Third  True  
freq           NaN      646    491   537  
mean     32.204208      NaN    NaN   NaN  
std      49.693429      NaN    NaN   NaN  
min       0.000000      NaN    NaN   NaN  
25%       7.910400      NaN    NaN   NaN  
50%      14.454200      NaN    NaN   NaN  
75%      31.000000      NaN    NaN   NaN  
max     512.329200      NaN    NaN   NaN

- survival rate only 38% while 62% did not survived

- half of the passanger were in third class

- most passanger were young adult -  median age =  26 year

- must be outlier in age shown 80 year old also

- median fare 14.45 but max was 512.33-it also may contain otliers most passanger travel alone

-most passanger embarked from 'S'

- most passanger class is Third Class

In [32]:
titanic.duplicated()

0      False
1      False
2      False
3      False
4      False
       ...  
886     True
887    False
888    False
889    False
890    False
Length: 891, dtype: bool

In [33]:
titanic.duplicated().sum()

118

In [34]:
titanic.drop_duplicates(inplace = True)

In [35]:
titanic.duplicated().sum()

0